src/nv_ingest/modules/sinks/vdb_task_sink.py

```diff
@@ -34,6 +39,61 @@
 
 VDBTaskSinkLoaderFactory = ModuleLoaderFactory(MODULE_NAME, MODULE_NAMESPACE, VdbTaskSinkSchema)
 
+_DEFAULT_ENDPOINT = os.environ.get("MINIO_INTERNAL_ADDRESS", "minio:9000")
+_DEFAULT_BUCKET_NAME = os.environ.get("MINIO_BUCKET", "nv-ingest")
+
+
+def _bulk_ingest(milvus_uri: str = None,
+                collection_name: str = None, 
+                bucket_name: str = None, 
+                bulk_ingest_path: str = None, 
+                extra_params: dict = None):
+
+    endpoint = extra_params.get("endpoint", _DEFAULT_ENDPOINT)
+    access_key = extra_params.get("access_key", None)
+    secret_key = extra_params.get("secret_key", None)
+
+    client = Minio(
+        endpoint,
+        access_key=access_key,
+        secret_key=secret_key,
+        session_token=extra_params.get("session_token", None),
+        secure=extra_params.get("secure", False),
+        region=extra_params.get("region", None),
+    )
+    bucket_found = client.bucket_exists(bucket_name)
+    if not bucket_found:
+        raise ValueError(f"Could not find bucket {bucket_name}")
+    batch_files = [[f"{file.object_name}"] for file in client.list_objects(bucket_name, prefix=bulk_ingest_path)]
+
+    uri_parsed = urlparse(milvus_uri)
+    conn = connections.connect(host=uri_parsed.hostname, port=uri_parsed.port)
+    collection = Collection(name=collection_name)
+
+    task_ids = []
+    for file in batch_files:
+        task_id = utility.do_bulk_insert(collection_name=collection_name, files=file)
+        task_ids.append(task_id)
+
+    while len(task_ids) > 0:
```

Comment:
If something goes wrong with a task here, it is removed just the same as if it completed successfully. This can take a undeterminable amount of time because it is dependent on the amount of data stored in the minio embedding directory that needs to be ingested.